Problem building on Ubuntu 20.04

Hi. Trying to build on a clean VM of Ubuntu 20.04. I'm building the latest stable (and master) and get the same.

```
[ 78%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/imgui_draw.cpp.o
[ 81%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/imgui_widgets.cpp.o
/root/parts/imhex/src/source/views/view_disassembler.cpp: In member function ‘virtual void hex::ViewDisassembler::createView()’:
/root/parts/imhex/src/source/views/view_disassembler.cpp:221:30: error: ‘CS_ARCH_M68K’ was not declared in this scope; did you mean ‘CS_ARCH_MAX’?
  221 |                         case CS_ARCH_M68K:
      |                              ^~~~~~~~~~~~
      |                              CS_ARCH_MAX
/root/parts/imhex/src/source/views/view_disassembler.cpp:222:30: error: ‘CS_ARCH_TMS320C64X’ was not declared in this scope
  222 |                         case CS_ARCH_TMS320C64X:
      |                              ^~~~~~~~~~~~~~~~~~
/root/parts/imhex/src/source/views/view_disassembler.cpp:223:30: error: ‘CS_ARCH_M680X’ was not declared in this scope; did you mean ‘CS_ARCH_MAX’?
  223 |                         case CS_ARCH_M680X:
      |                              ^~~~~~~~~~~~~
      |                              CS_ARCH_MAX
/root/parts/imhex/src/source/views/view_disassembler.cpp:224:30: error: ‘CS_ARCH_EVM’ was not declared in this scope; did you mean ‘CS_ARCH_ARM’?
  224 |                         case CS_ARCH_EVM:
      |                              ^~~~~~~~~~~
      |                              CS_ARCH_ARM
[ 84%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/imgui_demo.cpp.o
make[2]: *** [CMakeFiles/ImHex.dir/build.make:323: CMakeFiles/ImHex.dir/source/views/view_disassembler.cpp.o] Error 1
```

I have the following dependencies installed:
```
      - libmagic-dev
      - libglfw3-dev
      - libssl-dev
      - llvm-10-dev
      - nlohmann-json3-dev
      - libcapstone-dev
      - python3-dev
      - pkg-config
      - libglm-dev
      - gcc-10
      - g++-10
```

Perhaps I'm missing something? Capstone related? The only thing I modified was to bodge it to use Python 3.8.5 rather than the explicitly specified 3.8.6 as Ubuntu doesn't have that. But I don't think that's related.

Had the same issues on Ubuntu.  Posting here so hopefully helps other folks compiling on Ubuntu.

- Modified the CMakeLists.txt Python line to `find_package(Python 3.8 COMPONENTS Interpreter Development)`
- Built and installed Capstone 4.0.2 from source
- Called cmake: `CC="gcc-10" CXX="g++-10" cmake ..` to fix the `fatal error: concepts: No such file or directory 6 | #include <concepts>`

Now stuck at `/usr/bin/ld: cannot find -lLLVMDemangle`.  llvm is installed but guessing I'm not using a new enough version.

```
[ 88%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/imgui_demo.cpp.o
[ 91%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/imgui_impl_opengl3.cpp.o
[ 94%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/ImGuiFileBrowser.cpp.o
[ 97%] Building CXX object CMakeFiles/ImHex.dir/libs/ImGui/source/TextEditor.cpp.o
In file included from <redated>/opt/ImHex/source/lang/parser.cpp:3:
<redated>/opt/ImHex/include/helpers/utils.hpp: In instantiation of ‘std::string hex::format(const string&, Args ...) [with Args = {}; std::string = std::__cxx11::basic_string<char>]’:
<redated>/opt/ImHex/source/lang/parser.cpp:300:87:   required from here
<redated>/opt/ImHex/include/helpers/utils.hpp:25:32: warning: format not a string literal and no format arguments [-Wformat-security]
   25 |         ssize_t size = snprintf( nullptr, 0, format.c_str(), args ... );
      |                        ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<redated>/opt/ImHex/include/helpers/utils.hpp:31:17: warning: format not a string literal and no format arguments [-Wformat-security]
   31 |         snprintf(buffer.data(), size + 1, format.c_str(), args ...);
      |         ~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[100%] Linking CXX executable ImHex
/usr/bin/ld: cannot find -lLLVMDemangle
collect2: error: ld returned 1 exit status
make[2]: *** [CMakeFiles/ImHex.dir/build.make:595: ImHex] Error 1
make[1]: *** [CMakeFiles/Makefile2:77: CMakeFiles/ImHex.dir/all] Error 2
make: *** [Makefile:84: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -18,7 +18,8 @@ if(Python_VERSION LESS 3)
     message(FATAL_ERROR "No valid version of Python 3 was found.")
 endif()
 
-llvm_map_components_to_libnames(demangler)
+llvm_map_components_to_libnames(_llvm_libs demangle)
+llvm_expand_dependencies(llvm_libs ${_llvm_libs})
 
 include_directories(include ${GLFW_INCLUDE_DIRS} ${CAPSTONE_INCLUDE_DIRS} ${LLVM_INCLUDE_DIRS} libs/ImGui/include libs/glad/include ${Python_INCLUDE_DIRS})
 
@@ -83,10 +84,12 @@ add_executable(ImHex
         resource.rc
         )
 
+target_link_directories(ImHex PRIVATE ${LLVM_LIBRARY_DIR})
+
 if (WIN32)
-    target_link_libraries(ImHex libglfw3.a libgcc.a libstdc++.a libmagic.a libgnurx.a libtre.a libintl.a libiconv.a shlwapi.lib libcrypto.a libwinpthread.a libcapstone.a libLLVMDemangle.a ${Python_LIBRARIES} nlohmann_json::nlohmann_json)
+    target_link_libraries(ImHex libglfw3.a libgcc.a libstdc++.a libmagic.a libgnurx.a libtre.a libintl.a libiconv.a shlwapi.lib libcrypto.a libwinpthread.a libcapstone.a ${llvm_libs} ${Python_LIBRARIES} nlohmann_json::nlohmann_json)
 endif (WIN32)
 
 if (UNIX)
-    target_link_libraries(ImHex libglfw.so libmagic.so libcrypto.so libdl.so libcapstone.so libLLVMDemangle.so ${Python_LIBRARIES} nlohmann_json::nlohmann_json)
+    target_link_libraries(ImHex libglfw.so libmagic.so libcrypto.so libdl.so libcapstone.so ${llvm_libs} ${Python_LIBRARIES} nlohmann_json::nlohmann_json)
 endif (UNIX)

